src/Microsoft.Azure.SignalR.Common/Endpoints/AadAccessKey.cs

```diff
@@ -88,45 +97,52 @@ public virtual async Task<string> GenerateAadTokenAsync(CancellationToken ctoken
         internal void UpdateAccessKey(string kid, string accessKey)
         {
             Key = new Tuple<string, string>(kid, accessKey);
-            _lastUpdatedTime = DateTime.UtcNow;
-            _isAuthorized = true;
-            _initializedTcs.TrySetResult(null);
+            Authorized = true;
         }
 
-        internal async Task UpdateAccessKeyAsync()
+        internal async Task UpdateAccessKeyAsync(CancellationToken token = default)
         {
-            if (DateTime.UtcNow - _lastUpdatedTime < AuthorizeInterval)
+            var delta = DateTime.UtcNow - _lastUpdatedTime;
```

Comment:
is it possible to be called by multiple threads?

Reply:
No, there is only one `AccessKeySynchronizer` who will handle all the updates.

Reply:
And this method is `idempotent`.
It will work even if the method had been called twice or more times at once.